tanzu-build-service/build-image-action.md

```diff
@@ -0,0 +1,137 @@
+# GitHub Build Action for Tanzu Build Service (Alpha)
+
+This GitHub Action creates a Tanzu Build Service (TBS) Build on the given cluster.
+
+> **Important** Alpha features are experimental and are not ready for production use. Configuration and behavior are
+> likely to change and functionality may be removed in a future release.
+
+## Overview
+
+### Prerequisites
+- [TBS](install-tbs.hbs.md) 
+
+
+### Setup
+
+This action uses a Kubernetes Service Account with specific permissions.
+
+Here are the minimum required permissions needed:
+
+```yaml
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: github-actions
+rules:
+  - apiGroups: [ 'kpack.io' ]
+    resources:
+      - clusterbuilders
+    verbs: [ 'get' ]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: github-actions
+  namespace: dev
+rules:
+  - apiGroups: [ '' ]
+    resources: [ 'pods', 'pods/log' ]
+    verbs: [ 'get', 'watch', 'list' ]
+  - apiGroups: [ 'kpack.io' ]
+    resources:
+      - builds
+    verbs: [ 'get', 'watch', 'list', 'create', 'delete' ]
+```
+
+The action needs access to the cluster with the
+following [GitHub encrypted secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets) set:
+- CA_CERT
+- NAMESPACE
+- TOKEN
+- SERVER
+
+To get this information, run the following commands:
+
+```bash
+SECRET=$(kubectl get sa <sa-with-minimum-required-permissions> -oyaml -n <namespace-where-sa-exists> | yq '.secrets[0].name')
+CA_CERT=$(kubectl get secret $SECRET -oyaml -n <namespace-where-sa-exists> | yq '.data."ca.crt"')
+NAMESPACE=$(kubectl get secret $SECRET -oyaml -n <namespace-where-sa-exists> | ksd | yq .stringData.namespace)
+TOKEN=$(kubectl get secret $SECRET -oyaml -n <namespace-where-sa-exists> | ksd | yq .stringData.token)
+SERVER=$(kubectl config view --minify | yq '.clusters[0].cluster.server')
+```
+
+Create the required secrets on the repository
+through [GitHub.com](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository)
+or through the `gh` CLI:
+
+```bash
+gh secret set CA_CERT --app actions --body "$CA_CERT"
+gh secret set NAMESPACE --app actions --body "$NAMESPACE"
+gh secret set TOKEN --app actions --body "$TOKEN"
+gh secret set SERVER --app actions --body "$SERVER"
+```
+
+### Usage
+
+```yaml
+- uses: vmware-tanzu/build-image-action@v1-alpha
+  with:
+    ## Authorization
+    # Host of the API server
+    # If not running locally, use `${{ secrets.SERVER }}`
+    server: ''
+    # CA Certificate of the API Server
+    # If not running locally, use `${{ secrets.CA_CERT }}`
+    ca_cert: ''
+    # Service Account token to access kubernetes
+    # If not running locally, use `${{ secrets.TOKEN }}`
+    token: ''
+    # _(required)_ The namespace to create the build resource in
+    namespace:
+
+    ## Image configuration
+    # _(required)_ Destination for the built image
```

Comment:
is this required or optional? i thought tbs was configured with a default location for writing images which would make this optional?

also how do credentials for writing to for example the gcr registry get passed through in this case?

Reply:
At the moment this is a required field, we could look at querying the value for `buildservice.kp_default_repository` and construct a sensible default for the repo.

Credentials are configured in the TAP cluster. From the github side, the only thing we allow the user to configure is the name of the serviceAccount used to perform the build, that way, the credentials never leave the TAP cluster.

Reply:
I added some info on setting up the developer namespace - https://github.com/pivotal/docs-tap/pull/1927/files#diff-157384f69e31eab7b1a9eaea17d411ebf1bb7bc5398263dbee515fe07b4cb639R15